src/core/mac/mac.hpp

```diff
@@ -636,6 +668,29 @@ class Mac : public InstanceLocator
         kOperationTransmitOutOfBandFrame,
     };
 
+#if OPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_ENABLE
+    typedef struct
+    {
+        /**
+         * Histogram of number of retries for a single direct packet until success
+         * [0 retry: packet count, 1 retry: packet count, 2 retry : packet count ...
+         *  until max retry limit: packet count]
```

Comment:
With current implementation, the `0 retry` would include/count broadcast frames (which are always sent once by sub-mac layer). We need to be aware of this when the retry histogram is analyzed (e.g., if there are a lot of broadcast frame tx, it would make it look like a lot of frames were sent with 0 retry).

Alternative is to limit the retry histogram collection to unicast frames only (frames that require ack and can be retried).

Thoughts?

Reply:
@XPAWAN @LuDuda, any thoughts on this?

Reply:
The number of broadcast packets are counted by the TxBroadcast MAC counter and included in the histogram calculation.
I have tested broadcast frames and see that they are also re-transmitted when CSMACA operation fails.
The number of broadcast frames re-transmission is configured by mMaxFrameRetriesDirect MAC parameter (The same as Unicast frames).

Reply:
I'm okay with the proposed approach. Current TxRetry counter also counts retransmissions of broadcast frames triggered by CSMA/CA failures. So this implementation aligns with the current approach.